WordPress/Classes/ViewRelated/Stats/Extensions/Double+Stats.swift

```diff
@@ -32,31 +82,44 @@ extension Double {
     ///  - 1000000000000 becomes "1t"
 
     func abbreviatedString(forHeroNumber: Bool = false) -> String {
-        var num = self
-        let sign = num < 0 ? "-" : ""
-        num = fabs(num)
-
+        let absValue = fabs(self)
         let abbreviationLimit = forHeroNumber ? 100000.0 : 10000.0
 
-        if num < abbreviationLimit {
-            return "\(sign)\(num.formatWithCommas())"
+        if absValue < abbreviationLimit {
+            return self.formatWithCommas()
         }
 
-        let exp: Int = Int(log10(num) / 3.0)
-        let units: [String] = ["K", "M", "B", "T", "P", "E"]
-        let roundedNum: Double = Foundation.round(10 * num / pow(1000.0, Double(exp))) / 10
+        let exp: Int = Int(log10(absValue) / 3.0)
+        let unsignedRoundedNum: Double = Foundation.round(10 * absValue / pow(1000.0, Double(exp))) / 10
+
+        var roundedNum: Double
+        var unit: Unit
 
-        if roundedNum == 1000.0 {
-            return "\(sign)\(1)\(units[exp])"
+        if unsignedRoundedNum == 1000.0 {
+            roundedNum = 1
+            unit = units[exp]
         } else {
-            return "\(sign)\(roundedNum)\(units[exp-1])"
+            roundedNum = unsignedRoundedNum
+            unit = units[exp-1]
```

Comment:
It's probably impossible that `exp-1` will be out of bounds. But do you think we should still handle the possibility that it will in order to avoid a crash?

Reply:
Ah yea. Nice catch. Tis done!